File: C/enzoniko-cs50-problems-2019-x-mario-less/mario.c
//bibliotecas:
#include <cs50.h>
#include <stdio.h>
//função main:
int main(void){  
    //variáveis:
    int height, j, i;
    //pega a altura da pirâmide:
    printf("How tall is the Pyramid? ");
    scanf("%d",&height);
    //verifica se a altura esta entre 1 e 8 inclusive:
    while(height < 1){
        printf("How tall is the Pyramid? ");
        scanf("%d",&height);
    }
    while(height > 8){
        printf("How tall is the Pyramid? ");
        scanf("%d",&height);
    }
    while(height == 0){
        printf("How tall is the Pyramid? ");
        scanf("%d",&height);
    }
    //pega a altura e divide em andares:
    for ( i = 0; i < height; i++){
        //escreve os espaços necessários em cada andar:
        for (j = 0; j < height - i - 1; j++)
            printf(" ");
        //escreve os hashtags necessários em cada andar:
        for ( j = 0; j < i + 1; j++)
            printf("#");
        //divide os andares:
        printf("\n");
        
    } 
}

File: Python/enzoniko-cs50-problems-2019-x-sentimental-mario-less/mario.py
# Importa função "get_int" da biblioteca cs50
from cs50 import get_int
# Pede a altura da pirâmide
height = get_int("How tall is the Pyramid? ")
# Enquanto não for um valor entre 1 e 8 inclusive, repete a pergunta
while height < 1 or height > 8:
    height = get_int("How tall is the Pyramid? ")
# Pega o valor da altura e divide em andares
for i in range(height):
    # Escreve os espaços necessários
    for j in range(height - i - 1):
        print(" ", end="")
    # Escreve os blocos necessários
    for j in range(i + 1):
        print("#", end="")
    # Divide os andares
    print()